csharp/extractor/Semmle.Extraction.CIL/Entities/Instruction.cs

```diff
@@ -414,11 +419,17 @@ public override IEnumerable<IExtractionProduct> Contents
                         break;
                     case Payload.Arg8:
                     case Payload.Arg16:
-                        yield return Tuples.cil_access(this, Method.Parameters[(int)UnsignedPayloadValue]);
+                        if (!(Method.Parameters is null))
+                        {
+                            yield return Tuples.cil_access(this, Method.Parameters[(int)UnsignedPayloadValue]);
+                        }
```

Comment:
Another option is
```csharp
if (Method.Parameters is IList<Parameter> parameters)
    yield return Tuples.cil_access(this, parameters[(int)UnsignedPayloadValue]);
```
which results in only one call to `get_Parameters`.

Reply:
Calling `Method.Parameters` is not an expensive call here, is it? For readability, maybe `Method.Parameters is object` is the best as a `not null` check. What do you think?

Reply:
It is probably not expensive. I like `is object`; though it revealed a false positive in our `UselessTypeTest.ql` query.